server/src/service-layer/controllers/BookingController.ts

```diff
@@ -328,7 +334,11 @@ export class BookingController extends Controller {
         /** Adding the date and users to the response data array */
         responseData.push({
           date: slot.date,
-          users: combinedUsers
+          // Mapping the users to include the booking ID
+          users: combinedUsers.map((user, i) => ({
+            ...user,
+            bookingId: bookingSlotIds[i]
```

Comment:
We want to return the actual Id of the booking, instead of the booking slot id. Also a side note it's very brittle to be using indexes for something like this because you cant assume they are sorted. Would recommend the `find` method instead